stat/distuv/binomial.go

```diff
@@ -0,0 +1,181 @@
+// Copyright ©2017 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package distuv
+
+import (
+	"math"
+
+	"golang.org/x/exp/rand"
+
+	"gonum.org/v1/gonum/mathext"
+	"gonum.org/v1/gonum/stat/combin"
+)
+
+// Binomial implements the binomial distribution, a discrete probability distribution
+// that expresses the probability of a given number of successful Bernoulli trials
+// out of n total, each with successs probability p.
+// The poisson distribution has density function:
+//  f(k) = (n choose k) p^k (1-p)^(n-k)
+// For more information, see https://en.wikipedia.org/wiki/Binomial_distribution.
+type Binomial struct {
+	// N is the total number of Bernoulli trials. Must be greater than 0.
+	// P is the probablity of success in any given trial. Must be on [0, 1]
+	N, P float64
```

Comment:
N and P can be on separate lines. Then the comments refer directly to the fields.
Also, please rephrase "Must be greater than 0." to "N must be greater than 0." and "Must be on [0, 1]" to "P must be in [0, 1]."